lib/compress/zstd_compress.c

```diff
@@ -5659,6 +5702,13 @@ size_t ZSTD_compressStream2( ZSTD_CCtx* cctx,
             ZSTDMT_updateCParams_whileCompressing(cctx->mtctx, &cctx->requestedParams);
             cctx->cParamsChanged = 0;
         }
+        if (cctx->stableIn_notConsumed) {
+            assert(cctx->appliedParams.inBufferMode == ZSTD_bm_stable);
+            /* some early data was skipped - make it available for consumption */
+            assert(input->pos >= cctx->stableIn_notConsumed);
```

Comment:
This should also be a `RETURN_ERROR_IF()`.

Reply:
Same idea, `ZSTD_checkBufferStability()` is invoked just before this paragraph, and should already take care of user input issues. So if we nonetheless hit this `assert()` at this position, seems like it's a programmatic error.